crates/matrix-sdk-base/src/client.rs

```diff
@@ -376,16 +376,20 @@ impl BaseClient {
                         let actions = push_rules.get_actions(&event.event, context).to_vec();
```

Comment:
Can we guard this by the sender check as well? It's a non-trivial operation.

Also the sender check can be simplified to

```rust
if event.event.get_field::<OwnedUserId>("sender")?.map_or(false, |id| id == user_id)
```

to avoid deserializing the whole event.